gamemode/modules/commands/sv_commands.lua

```diff
@@ -0,0 +1,290 @@
+
+    return num
+end )
+
+commands.addType( "float", function( arg )
+    local num = tonumber( arg )
+
+    if not num then
+        return "Invalid float: " .. arg
+    end
+
+    return num
+end )
+
+local booleanValues = {
+    ["enabled"]  = true,
+    ["enable"]   = true,
+    ["true"]     = true,
+    ["yes"]      = true,
+    ["1"]        = true,
+
+    ["disabled"] = false,
+    ["disable"]  = false,
+    ["false"]    = false,
+    ["no"]       = false,
+    ["0"]        = false
+}
+
+commands.addType( "bool", function( arg )
+    if booleanValues[arg] ~= nil then
+        return booleanValues[arg]
+    end
+
+    return nil, "Invalid boolean: " .. arg
+end )
+
+commands.addType( "string", function( arg )
+    return arg
+end )
+
+commands.addType( "player", function( arg, ply )
+    if arg == "^" then
+        return ply
+    end
+
+    if arg == "@" then
+        local target = ply:GetEyeTrace().Entity
+        local isValidPlayer = IsValid( target ) and target:IsPlayer()
+
+        if isValidPlayer then
+            return target
+        end
+
+        return nil, "Not currently aiming at a player!"
+    end
+
+    local isSteamID = string.find( arg, "STEAM_" )
+
+    if isSteamID then
+        local ply = player.GetBySteamID( arg )
+
+        if ply then
+            return ply
+        else
+            return nil, "Invalid player: " .. arg
+        end
+    end
+
+    local selectedPlayers = {}
+    local playerList = ""
+
+    arg = string.lower( arg )
+
+    for _, ply in pairs( player.GetAll() ) do
+        local plyNick = string.lower( ply:Nick() )
+
+        if string.find( plyNick, arg ) then
+            table.insert( selectedPlayers, ply )
+            playerList = playerList .. "\n" .. plyNick
+        end
+    end
+
+    if #selectedPlayers > 1 then
+        return nil, "\"" .. arg .. "\" matches multiple players:" .. playerList
+    elseif #selectedPlayers == 1 then
+        return selectedPlayers[1]
+    end
+
+    return nil, "Invalid player: " .. arg
+end )
+
+function commands.checkArguments( argNames, argTypes, args, ply )
+    if #args < #argTypes then
+        local commandHelp = ""
+
+        for i, argName in ipairs( argNames ) do
+            local argType = argTypes[i]
+
+            commandHelp = commandHelp .. argName .. ":" .. argType .. " "
+        end
+
+        return nil, "Missing argument. Usage " .. commandHelp
+    end
+
+    local newArgs = {}
+
+    for i, argType in ipairs( argTypes ) do
+        local arg = args[i]
+
+        local value, errorMsg = commands.types[argType]( arg, ply )
+
+        if errorMsg then
+            return nil, errorMsg
+        end
+
+        newArgs[i] = value
+    end
+
+    return newArgs
+end
+
+local function processArguments( argsStr )
+    argsStr = string.Trim( argsStr )
+
+    local args = {}
+    local str = ""
+
+    local insideQuotes = false
+
+    local i = 1
+
+    while i <= string.len( argsStr ) do
+        local char = string.sub( argsStr, i, i )
```

Comment:
i dont like this function being duplicated. could it be defined in the shared realm? you could have a boolean argument keepQuotes? and if you are duplicating them you could do `argsStr[i]` in this function as well so they  are consistent

Reply:
They both already have argsStr[i] from a recent push but I like the idea of having an argument for keeping quotes since it'S the only difference. I'll work on that